### PR TITLE
Fixes #1240 #1259 - Put label editor on right side of issues page

### LINKED ISSUE #1240: Put label editor on right side of issues page.
Currently it's down by the comment widget, which is great if you're going to comment. Not so great if you want to see what labels are currently applied.

We'd also like to put a "milestone/status/whatever" widget under (or above?) the labels widget.

cc @magsout

### PR DESCRIPTION
**Pull request by magsout**
OK, let's try this to fixes #1240 and #1259 

screen recording 2017-03-04 at 12 01 am

First step: I put the label editor and some informations on the right side.
For DOM reason and JS, the label editor is under the title on small screen. We need to think about this label Editor on small screen after this PR or maybe during this PR.

We ( @miketaylr ) need to add some JS about this new aside.

But at first, tell me what do you think about this ?

@karlcow @adamopenweb @zoepage @miketaylr @denschub

### CODE REVIEW

**Comment by miketaylr** [comment]
> But at first, tell me what do you think about this ?

I think it looks awesome, @magsout. Excited for it to land.

> We ( @miketaylr ) need to add some JS about this new aside.

Happy to help out -- what specifically needs to be done?

**Comment by karlcow** [comment]
@magsout This is improving the UX a lot. ❤️ 
And agreed with you for the mobile version on how we can make it better. 
We are reaching step by step a good solution. Politique des petits pas. 👣 

And as usual, incremental shipping. Let's not try to fix everything here.

**Comment by magsout** [comment]
thanks you all for your feedbkack.

@miketaylr let me push a new commit and I come back for you.
@karlcow yes, agree with you.


One question about `sticky` do we add a Polyfill? See the browser's support http://caniuse.com/#search=sticky

**Comment by magsout** [comment]
@miketaylr this file https://github.com/webcompat/webcompat.com/pull/1389/files#diff-6f2a882bffbedc27676bc95ced4f8ae9 isn't dynamic. I have split the original template in two. I don't know if I have to create another template in backbone or if I have to do .. what?

**Comment by miketaylr** [comment]
> One question about sticky do we add a Polyfill? See the browser's support http://caniuse.com/#search=sticky

Hmm. That's probably a good idea.

**Comment by miketaylr** [comment]
> @miketaylr this file https://github.com/webcompat/webcompat.com/pull/1389/files#diff-6f2a882bffbedc27676bc95ced4f8ae9 isn't dynamic. I have split the original template in two. I don't know if I have to create another template in backbone or if I have to do .. what?

Ah yeah. It needs a backbone view and template. I can put that together unless you'd like to take a stab!

**Comment by magsout** [comment]
@miketaylr I tried something, but it doesn't work.. If you can enlighten me

**Comment by miketaylr** [comment]
I'll take a look this afternoon or in the morning -- thanks!

**Comment by miketaylr** [comment]
> I'll take a look this afternoon or in the morning -- thanks!

Uhhhh, and then I went on vacation. This is on my TODO list for this week.

**Comment by magsout** [comment]
> I'll take a look this afternoon or in the morning -- thanks!

>Uhhhh, and then I went on vacation. This is on my TODO list for this week.

❤

**Comment by miketaylr** [comment]
OK, this should be working now. I'll point out in the code what I had to do, and have one question for you @magsout.

File: webcompat/static/js/lib/issues.js [hunk 1]
```diff
@@ -374,6 +388,7 @@ issues.MainView = Backbone.View.extend({
     var issueModel = {model: this.issue};
     this.metadata = new issues.MetaDataView(issueModel);
     this.body = new issues.BodyView(_.extend(issueModel, {mainView: this}));
+    this.aside = new issues.AsideView(issueModel);
```
Comment (miketaylr) [inline webcompat/static/js/lib/issues.js]: The `AsideView` just needed to be constructed (and passed a model) to render.

File: webcompat/templates/issue/issue-aside.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          Repoter:
+          <span class="wc-Issue-reporter">
+            <a class="wc-Link" href="https://github.com/<%= reporter %>"><%= reporter %></a>
+          </span>
+        </div>
+        <div>Comments: <%= commentNumber %></div>
+      </div>
+    </script>
+  </div>
+  <div class="wc-Issue-information-content">
```
Comment (miketaylr) [inline webcompat/templates/issue/issue-aside.html]: @magsout it seems like backbone wants to ignore any other non-template DOM inside the templates parent. So, I had to move the `.wc-Issue-information-content .wc-Issue-labels` stuff to be a sibling. Maybe something we could tweak or fix with a wrapper? 🤔

File: webcompat/static/js/lib/issues.js (hunk 1 above)
Comment (magsout) [inline webcompat/static/js/lib/issues.js]: ahhh, ok ... thanks

File: webcompat/templates/issue/issue-aside.html (hunk 2 above)
Comment (magsout) [inline webcompat/templates/issue/issue-aside.html]: ah yes good catch.

**Comment by magsout** [comment]
@miketaylr thanks a lot for your help.

Ready for review ;)

**Comment by miketaylr** [comment]
> Ready for review ;)

\o/

File: webcompat/static/css/development/components/issue.css [hunk 3]
```diff
@@ -139,16 +135,17 @@
...
 .wc-Issue-labels {
-  margin-bottom: 2em;
+  background-color: #fff;
+  margin: 2px 0;
+  padding: 1em;
 }
 /* FullScreen LabelEditor*/
 @media all and (max-width: 23.43em) {
-
-  .wc-Issue-lbels {
```
Comment (miketaylr) [inline webcompat/static/css/development/components/issue.css]: oops, good catch.

File: webcompat/static/css/development/components/issue.css [hunk 4]
```diff
@@ -164,15 +161,31 @@
...
+@media all and (min-width: 48.13em) {
+  .wc-Issue-aside {
+    float: right;
+    width: 30%;
+  }
+}
+
+@media all and (min-height: 40em), all and (min-width: 48.13em) {
+  .wc-Issue-aside {
+    position: sticky;
```
Comment (miketaylr) [inline webcompat/static/css/development/components/issue.css]: Just tested in Edge, and this degrades to `position: static` nicely. 👍

File: webcompat/static/css/development/components/label.css [hunk 5]
```diff
@@ -1,20 +1,32 @@
...
+    .wc-Label {
+      display: block;
+      padding: .375rem .625rem;
+    }
+  }
+
+  @media all and (max-width: 48.12em) {
+    .wc-Label {
+      display: inline-block;
+      line-height:1;
```
Comment (miketaylr) [inline webcompat/static/css/development/components/label.css]: nit: missing space `line-height:1;`

**Comment by miketaylr** [comment]
Ah wait, we need to fix one test, I think: 
`NoSuchElement: [POST http://localhost:4444/wd/hub/session/04d70740-de27-4c33-a612-f633ef5cda14/element / {"using":"css selector","value":".js-Issue-reporter"}] Unable to locate element: .js-Issue-reporter`

File: webcompat/templates/issue/issue-information.html [hunk 6]
```diff
@@ -1,26 +1,9 @@
...
-    <header class="wc-Issue-information-header wc-Issue-information-header--<%= stateClass %>">
-      <div><%= issueState %></div>
-      <div>#<%= number %></div>
-    </header>
-    <div class="wc-Issue-information-content">
-      <h1 class="wc-Issue-title js-Issue-title">
-        <%= title %>
-      </h1>
-      <div class="wc-Issue-date">Opened <%= createdAt %> by
-        <span class="wc-Issue-reporter js-Issue-reporter">
```
Comment (miketaylr) [inline webcompat/templates/issue/issue-information.html]: Yeah, the test relied on this class.

File: webcompat/templates/issue/issue-aside.html [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      <div class="wc-Issue-information-content">
+        <h1 class="wc-Issue-title js-Issue-title">
+          <%= title %>
+        </h1>
+      </div>
+      <div class="wc-Issue-information-content">
+        <div>Opened: <%= createdAt %></div>
+        <div>
+          Repoter:
+          <span class="wc-Issue-reporter">
```
Comment (miketaylr) [inline webcompat/templates/issue/issue-aside.html]: We just need to add the `js-Issue-reporter` class to this span to get the test to pass.

File: webcompat/static/css/development/components/issue.css (hunk 4 above)
Comment (magsout) [inline webcompat/static/css/development/components/issue.css]: @miketaylr don't know if a polyfill is necessary. Chrome, Safari and Firefox support sticky, but not edge, what do you think?

**Comment by magsout** [comment]
cool, tests have passed successful

**Comment by magsout** [comment]
hum, now tests have failed :( ... just squash some commit and pushed..

**Comment by magsout** [comment]
So, test have passed successfull.

A little weird, because I have to restart build on travis 3 times.. Need to check the log to understand why. But maybe There are issue see https://github.com/webcompat/webcompat.com/issues/1470

File: webcompat/static/css/development/components/issue.css (hunk 4 above)
Comment (miketaylr) [inline webcompat/static/css/development/components/issue.css]: Nope, no polyfill needed. I tested in Edge, and it behaves fine enough. It doesn't seem broken.

**Comment by miketaylr** [comment]
> A little weird, because I have to restart build on travis 3 times..

Yeah, the tests are flaky. We'll have someone working on that this summer. 🎉 🍰

**Comment by miketaylr** [comment]
This LGTM, ready to merge.

Thanks a billion @magsout! ❤️ 💟 😻

**Pull request merged**